### ISSUE TITLE
ECDSA root cert

### ISSUE THREAD

**Issue by budulinek** [hakwerk/labca#46]
Hi,
many thanks for this great project! I have been looking for private CA for my home network and LabCa looks like perfect solution. Though I have an issue with ECDSA root cert.

I have a clean install on Debian 11 (virtual machine). Choose ECDSA-384 when generating root cert during setup. Generate RSA-2048 issuer cert. Both certs are correctly generated. After restart, the setup process fails during an attempt to generate cert for the webserver. Boulder throws an error:

`boulder-boulder-1  | E220602215402 boulder-ca qM-tDQA [AUDIT] Couldn't load issuers: failed to create lint issuer: x509: requested SignatureAlgorithm does not match private key type`

Looking at the certs:
`/home/labca/admin/data/root-ca.pem`
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (384 bit)
    Signature Algorithm: ecdsa-with-SHA384

`/home/labca/admin/data/issuer/ca-int.pem`
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
    Signature Algorithm: ecdsa-with-SHA384

And the key + request generated by acme_tiny.py
`/home/labca/nginx_data/ssl/account.key`
Key Details:
   Type: RSA
   Size (bits): 4096

`/home/labca/nginx_data/ssl/domain.csr`
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
    Signature Algorithm: sha256WithRSAEncryption

I am not an expert but IMHO the problem is the private key type and/or signature algorithm (sha256) used by acme_tiny.py in the request. 
What do you think? 
Thanks a lot!

**Comment by hakwerk** [comment]
The problem does not seem to be in acme_tiny.py but in the Let's Encrypt boulder core: it does not like a combination of RSA and ECDSA for the issuer and the root CA.

In the past there was another issue with ECDSA issuers so that's why currently my GUI limits the issuer to RSA, but it looks like that has been resolved now. I tested with ECDSA for both the root and the issuer and that could sign the RSA webserver cert just fine.

I'll investigate further if I can find and fix the limitation in boulder, otherwise I'll change my GUI to only allow the same algorithm for the root and the issuer.
